src/v2/Components/FlashBanner/index.tsx

```diff
@@ -82,18 +83,21 @@ const TrackedFlashBanner = track({
 })(FlashBanner)
 
 export const FlashBannerQueryRenderer: React.FC = () => {
-  const { relayEnvironment } = useSystemContext()
+  const { relayEnvironment, user } = useSystemContext()
 
   return (
     <SystemQueryRenderer
       environment={relayEnvironment}
-      query={graphql`
-        query FlashBannerQuery {
-          me {
-            canRequestEmailConfirmation
+      query={
+        user &&
+        graphql`
```

Comment:
This is where the magic happens.

Reply:
This is pretty unconventional in terms of how we work with GraphQL / Relay. If there's no user, we don't need to execute SystemQueryRenderer at all:

```ts
user ? <ComponentA /> : <ComponentB />
```